composites/Plugin/ContentAnalysis/components/ContentAnalysis.js

```diff
@@ -132,15 +140,26 @@ class ContentAnalysis extends React.Component {
 	getResults( results ) {
 		return results.map( ( result ) => {
 			let color = this.getColor( result.rating );
+			let isPressed = result.id === this.state.checked;
+			let ariaLabel = "";
+			if ( this.props.marksButtonStatus === "disabled" ) {
```

Comment:
I would place most of this if-else statement outside of the mapping:
```
if ( this.props.marksButtonStatus === "disabled" ) {
   defaultAriaLabel = this.props.intl.formatMessage( messages.disabledButton );
} else {
   defaultAriaLabel = this.props.intl.formatMessage( messages.highlight );
}
```

If you put this before the return then you can do the following in the map callback:
```
let ariaLabel = defaultAreaLabel;
if ( this.props.marksButtonStatus === "disabled" ) {
    ariaLabel = this.props.intl.formatMessage( messages.noHighlight );
}
```

Reply:
Those three ifs are too intertwined to do that